### PR TITLE
Challenge/3/state management

### PR DESCRIPTION
**Pull request by Gonzz-96**
### What does this PR do?

Create `ThemeContext` and `KeywordContext` to have both as global context in the application. They both use the `useReducer` hook to manage their changes.

### CODE REVIEW

File: src/context/SearchContext.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    case 'RESET':
+      return 'wizeline';
+    default:
+      throw new Error('Ups, action not supported');
+  }
+};
+
+export default KeywordContext;
+
+export { reducer };
```
Comment (PacoMojica) [inline src/context/SearchContext.js:18]: I like this reducer. specially the error message 😆

File: src/context/ThemeContext.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+const darkTheme = {
+  isDarkMode: true,
+  color: '#FFFFFF',
+  backgroundColor: '#303030',
+  cardBackgroundColor: '#424242',
+  cardTextColor: '#C7C7C7',
+};
+
+const ThemeContext = React.createContext({});
```
Comment (PacoMojica) [inline src/context/ThemeContext.js:18]: here you can use a theme from the ones you defined previously, for example:

```jsx
const ThemeContext = React.createContext(lightTheme);
```

**Review by PacoMojica — COMMENTED** [review]
Hi Gonzz, I like what you have up to this point, as you said the only thing missing are the tests for the providers, I'd like to see at least one test for any of the providers that you have. Great work!

File: src/context/ThemeContext.js (hunk 2 above)
Comment (Gonzz-96) [reply src/context/ThemeContext.js:18]: @PacoMojica, I used that, but for some reason, when I retrieved the context in another component, it was null. Is there a reason this happened?

**Pull request merged**